frontend/src/components/TradeBox/EncryptedChat/index.tsx

```diff
@@ -1,522 +1,56 @@
-import React, { useEffect, useState } from 'react';
-import { useTranslation } from 'react-i18next';
-import {
-  Button,
-  IconButton,
-  Tooltip,
-  TextField,
-  Grid,
-  Container,
-  Card,
-  CardHeader,
-  Paper,
-  Typography,
-} from '@mui/material';
-import { encryptMessage, decryptMessage } from '../../../pgp';
-import { saveAsJson } from '../../../utils';
-import { AuditPGPDialog } from '../../Dialogs';
-import RobotAvatar from '../../RobotAvatar';
-import { systemClient } from '../../../services/System';
-import { websocketClient, WebsocketConnection } from '../../../services/Websocket';
-
-// Icons
-import CheckIcon from '@mui/icons-material/Check';
-import CloseIcon from '@mui/icons-material/Close';
-import ContentCopy from '@mui/icons-material/ContentCopy';
-import VisibilityIcon from '@mui/icons-material/Visibility';
-import CircularProgress from '@mui/material/CircularProgress';
-import KeyIcon from '@mui/icons-material/Key';
-import { ExportIcon } from '../../Icons';
-import { useTheme } from '@mui/system';
+import React, { useState } from 'react';
+import EncryptedSocketChat from './EncryptedSocketChat';
+import EncryptedTrutleChat from './EncryptedTrutleChat';
 
 interface Props {
+  turtleMode: boolean;
   orderId: number;
+  takerNick: string;
+  makerNick: string;
   userNick: string;
+  chatOffset: number;
 }
-
-interface EncryptedChatMessage {
+export interface EncryptedChatMessage {
   userNick: string;
   validSignature: boolean;
   plainTextMessage: string;
   encryptedMessage: string;
   time: string;
   index: number;
 }
+export interface ServerMessage {
+  message: string;
+  time: string;
+  index: number;
+  nick: string;
+}
 
-const EncryptedChat: React.FC<Props> = ({ orderId, userNick }: Props): JSX.Element => {
-  const { t } = useTranslation();
-  const theme = useTheme();
-
-  const audio = new Audio(`/static/assets/sounds/chat-open.mp3`);
-  const [connected, setConnected] = useState<boolean>(false);
-  const [peerConnected, setPeerConnected] = useState<boolean>(false);
-  const [ownPubKey] = useState<string>(
-    (systemClient.getCookie('pub_key') ?? '').split('\\').join('\n'),
-  );
-  const [ownEncPrivKey] = useState<string>(
-    (systemClient.getCookie('enc_priv_key') ?? '').split('\\').join('\n'),
-  );
-  const [peerPubKey, setPeerPubKey] = useState<string>();
-  const [token] = useState<string>(systemClient.getCookie('robot_token') || '');
+const EncryptedChat: React.FC<Props> = ({
+  turtleMode,
+  orderId,
+  takerNick,
+  userNick,
+  chatOffset,
+}: Props): JSX.Element => {
   const [messages, setMessages] = useState<EncryptedChatMessage[]>([]);
```

Comment:
By keeping messages here, they we'll be shared between both components and switching  won't affect to the UX, if not the chant went totally blank because the component were loading entirely